splinterd/src/daemon.rs

```diff
@@ -982,6 +984,35 @@ fn start_health_service(
 fn create_store_factory(
     db_url: &str,
 ) -> Result<Box<dyn splinter::store::StoreFactory>, StartError> {
+    // If the given db_url is a file path check that the file exists and is not empty
```

Comment:
The logic here is similar to that which happens in db_url.parse() in the following statement, which determines the type of database URI which has been provided (postgres or sqlite). It is probably easier to maintain consistency if the  check is done against the connection_uri (and only for sqlite variant).

It may make sense to place this check within create_store_factory() to ensure that it it is not possible to create a factory which accidentally creates empty files.

Reply:
Moved this to `create_store_factory()`